### DISCUSSION TITLE
Beams on new layer

### DISCUSSION THREAD

**Discussion by hutzlibu** [damienBloch/inkscape-raytracing#3] [Ideas]
Hey there, 

nice plugin.
What would improve working with it for me, would be, if all beams would be added on a new layer, instead of the same as the lenses etc. Then I could faster see the result and go back and modify.
Alternatively a button to remove all Beams by click.
(Strg + Z works for now, though)

Also, hotkeys would be helpful, for faster editing (instead of clicking).

It seams you can edit those in:
`share/inkscape/keys/inkscape.xml`

And they are in the shape 
`<bind key="q" modifiers="Alt" action="EffectLast" display="true" />`
for example, 

So I need to know the exact name of the "action" for Ray Tracing for example

`<bind key="E" modifiers="Alt" action="RayTracing" display="true" />`
 
Does not work sadly.

(Side note: I could not get it to work on windows, it just failed silently and I could not get Inkscape to show me some log, but on Manjaro Linux it all worked without problem,
that means, alert showed me that the lxml module was missing, which I just installed with 
pacman -S python-lxml
)

**Comment by damienBloch** [comment]
Hi,

Thanks for the feedback! 

Having a better way to clean the beams is a good idea. I think I'll add them to a new layer and maybe also add an option to remove all the beams in a selection.

For the hotkeys, it's possible to directly bind an extension with Edit/Preferences/Interface/Keyboards shortcuts/Extensions tab (I've checked on Inkscape1.1, not 100% sure if it exists on previous versions).

I'v tried on windows and it worked without problem. It doesn't seem to be a python dependency issue on your side, otherwise it would have probably just complained explicitly, so not sure what the issue could be.
